operator/pkg/reconciliation/reconcile_racks.go

```diff
@@ -703,6 +702,32 @@ func (rc *ReconciliationContext) createUser(user api.CassandraUser) error {
 	return err
 }
 
+func (rc *ReconciliationContext) GetUsers() []api.CassandraUser {
+	dc := rc.Datacenter
+	// add the standard superuser to our list of users
+	users := dc.Spec.Users
+	users = append(users, api.CassandraUser{
+		Superuser: true, 
+		SecretName: dc.GetSuperuserSecretNamespacedName().Name,
+	})
+
+	return users
+}
+
+func (rc *ReconciliationContext) UpdateSecretWatches() error {
+	dc := rc.Datacenter
+	users := rc.GetUsers()
+	names := []types.NamespacedName{}
+	for _, user := range users {
+		name := types.NamespacedName{Name: user.SecretName, Namespace: dc.Namespace}
+		names = append(names, name)
+	}
+	dcNamespacedName := types.NamespacedName{Name: dc.Name, Namespace: dc.Namespace,}
+	err := rc.SecretWatches.UpdateWatch(dcNamespacedName, names)
```

Comment:
As an aside, if we wanted to, we could implement the dynamic watches without using annotations. Thats because this call here in reconciliation provides us with all the state we need, so it need not be persisted someplace else.

Reply:
Hmmm.  Would that be a win, or does the annotation buy us anything?

Reply:
Jim brought up that we really don't do anything like that at present in the operator.

Reply:
yeah did we see examples of any other operators building up state in memory?

Reply:
Yes, elastic operator does this all in memory. It's a bit convoluted (because they have other stuff going on) but it is all in memory.

Reply:
Specifically: https://github.com/elastic/cloud-on-k8s/blob/master/pkg/controller/common/watches/handler.go

It's a bit harder to follow, but they do the same thing, except providing individual event handlers, whereas I'm returning the NamespacedName to use when queueing a reconciliation request (which I think is sufficient for our needs). They update these registrations during reconciliation just like we do.